### PR TITLE
nixos/zfs: Fix boot.zfs.forceImportAll

### PR DESCRIPTION
**Pull request by srhb**
According to the docs of [boot.zfs.forceImportAll](https://nixos.org/nixos/options.html#forceimportall) this option makes it so that non-root pools are forcibly imported. As far as I can figure out, and as the test included in this PR demonstrates, it actually does nothing :goggles:

Fix it so that the zpool pool import service forcibly imports if the option is `true` (the default.)

cc @adisbladis maybe? :slightly_smiling_face:

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by numinit** [comment]
@asdf8dfafjk Looks like it's enabled by default for backwards compatibility reasons? Shouldn't be a huge deal, I don't think.

**Review by numinit — APPROVED** [review]
Looks like a pretty simple change. I think this more closely matches intended behavior, so +1 to this.

**Comment by asdf8dfafjk** [comment]
You guys are much more experienced but the "new backward compatibility" is that  by default non-root pools are not forcefully imported. I'll defer to you guys anyway.

**Comment by srhb** [comment]
@asdf8dfafjk I agree with you. It used to be false, and now it would be true. I will fix this once I'm back from vacation. It should continue being false.

**Comment by numinit** [comment]
I'm good with it either way.

**Comment by srhb** [comment]
@asdf8dfafjk I have changed the default so that it matches the previous behaviour. Thanks for the diligence. :)

**Comment by jonringer** [comment]
do you mind writing in the 2103 release notes that this is a potential breaking change for some users

File: nixos/doc/manual/release-notes/rl-2103.xml
Comment (kisik21) [inline nixos/doc/manual/release-notes/rl-2103.xml:67]: "forcibly imported" might sound a little bit better, but the documentation itself is clear enough and understandable. I had to read a little bit of ZFS documentation to understand why this might even be needed and what safeguards are bypassed by this option, but, well, we all learn something new every day, don't we?

**Review by vikanezrimaya — APPROVED** [review]
LGTM.

Also off-topic, but: can I just say that I love to read our new Python-based tests? :smile_cat: 

/status needs_merger

File: nixos/doc/manual/release-notes/rl-2103.xml
Comment (kevincox) [reply nixos/doc/manual/release-notes/rl-2103.xml:67]: I agree with this. I'll merge once this is updated.

Comment (srhb) [reply nixos/doc/manual/release-notes/rl-2103.xml]: Thanks, I've changed the wording. I understand the concern wrt. having to read ZFS documentation, but I'm a bit wary of including too much here, so I didn't make any other changes for now. :)

**Pull request merged**
